bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/util/ShellyChannelCache.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    /**
+     * Update one channel. Use Channel Cache to avoid unnecessary updates (and avoid
+     * messing up the log with those updates)
+     *
+     * @param channelId Channel id
+     * @param value Value (State)
+     * @param forceUpdate true: ignore cached data, force update; false check cache of changed data
+     * @return true, if successful
+     */
+    @SuppressWarnings("null")
+    public boolean updateChannel(String channelId, State value, Boolean forceUpdate) {
+        Validate.notNull(channelData);
+        Validate.notNull(channelId);
+        Validate.notNull(value, "updateChannel(): value must not be null!");
+        try {
+            Object current = channelData.get(channelId);
+            if (!enabled || forceUpdate || (current == null) || !current.equals(value)) {
+                // For channels that support multiple types (like brightness) a suffix is added
+                // this gets removed to get the channelId for updateState
+                thingHandler.publishState(channelId, value);
+                synchronized (channelData) {
+                    if (current == null) {
+                        channelData.put(channelId, value);
+                    } else {
+                        channelData.replace(channelId, value);
+                    }
+                }
+                logger.trace("{}: Channel {} updated with {} (type {}).", thingName, channelId, value,
+                        value.getClass());
+                return true;
+            }
+        } catch (IllegalArgumentException | NullPointerException e) {
+            logger.debug("{}: Unable to update channel {} with {} (type {}): {} ({})", thingName, channelId, value,
+                    value.getClass().toString(), ShellyUtils.getString(e), e.getClass().toString());
+        }
+        return false;
+    }
+
+    public boolean updateChannel(String group, String channel, State value) {
+        return updateChannel(mkChannelId(group, channel), value, false);
+    }
+
+    public boolean updateChannel(String channelId, State value) {
+        return updateChannel(channelId, value, false);
+    }
+
+    /**
+     * Get a value from the Channel Cache
+     *
+     * @param group Channel Group
+     * @param channel Channel Name
+     * @return the data from that channel
+     */
+    @Nullable
+    public Object getValue(String group, String channel) {
+        return getValue(mkChannelId(group, channel));
+    }
+
+    @Nullable
+    public Object getValue(String channelId) {
+        synchronized (channelData) {
+            return channelData.get(channelId);
+        }
+    }
+
+    public void resetChannel(String channelId) {
+        Validate.notNull(channelId);
+        synchronized (channelData) {
+            if (channelData.containsKey(channelId)) {
+                channelData.remove(channelId);
+            }
+        }
+
+    }
+
+    public void clear() {
+        channelData = new HashMap<>();
```

Comment:
No need to be wasteful.
```suggestion
        channelData.clear();
```

Reply:
changed